src/main/java/com/zendesk/maxwell/row/RowMap.java

```diff
@@ -282,6 +320,25 @@ private String jsonFromStream() {
 		return s;
 	}
 
+	private String encryptedJsonFromStream(JsonGenerator g, MaxwellOutputConfig outputConfig) throws IOException, NoSuchAlgorithmException {
+		SecureRandom randomSecureRandom = SecureRandom.getInstance("SHA1PRNG");
+		byte[] iv = new byte[16];
+		randomSecureRandom.nextBytes(iv);
+		byte[] init_vector = iv;
+		ByteArrayOutputStream outputStream = byteArrayThreadLocal.get();
+		String encryptedJSON = RowEncrypt.encrypt(outputStream.toString(), outputConfig.secret_key, init_vector);
+		outputStream.reset();
+		g.writeStartObject();
+		g.writeStringField("data", encryptedJSON);
+		g.writeStringField("init_vector", Base64.encodeBase64String(init_vector));
+		g.writeEndObject();
+		g.flush();
+
+		String s = outputStream.toString();
+		outputStream.reset();
+		return s;
+	}
+
```

Comment:
This method duplicates a lot of its code with `EncryptingJsonGenerator` - can you reuse that class instead of duplicating?